### PR TITLE
Documentation site update

### PR DESCRIPTION
**Pull request by abhro**
- Use Documenter.jl v1
- Clean installation instructions
- Add headers for docstrings
- Use `Downloads.download` instead of `Base.download`
- Add cross-reference links

### CODE REVIEW

File: src/FileIO.jl [hunk 1]
```diff
@@ -11,6 +11,7 @@ export DataFormat,
        add_saver,
        filename,
        file_extension,
+       info,
```
Comment (nhz2) [inline src/FileIO.jl:14]: Why is `info` being exported in this PR?

Comment (abhro) [reply src/FileIO.jl:14]: `info` is already listed in the `FileIO` module docstring, so I added a link and the docstring to the documentation site. The export is just to make more sure that it's part of the public API.

Comment (nhz2) [reply src/FileIO.jl:14]: Other people may have opinions on exporting `info` so it would be good to make that change in a separate PR.

Comment (abhro) [reply src/FileIO.jl:14]: That's fair. I'll push a revert soon. Thanks!

Comment (abhro) [reply src/FileIO.jl]: Reverted in [a33d30d](https://github.com/JuliaIO/FileIO.jl/pull/393/commits/a33d30df84639e3fd312afc0a2e426b9a1e2d732)!

**Comment by nhz2** [comment]
Thank you for fixing this up

**Pull request merged**